.github/workflows/functional-test.yaml

```diff
@@ -576,6 +576,29 @@ jobs:
         with:
           name: ${{ matrix.name }}_container_logs
           path: ./${{ env.RADIUS_CONTAINER_LOG_BASE }}
+      - name: Get Terraform recipe publishing logs
+        if: always()
+        run: |
+          # Create pod-logs directory
+          mkdir -p recipes/pod-logs
+          # Get pod logs and save to file
+          namespace="radius-test-tf-module-server"
+          label="app.kubernetes.io/name=tf-module-server"
+          pod_names=($(kubectl get pods -l $label -n $namespace -o jsonpath='{.items[*].metadata.name}'))
+          for pod_name in "${pod_names[@]}"; do
+            kubectl logs $pod_name -n $namespace > recipes/pod-logs/${pod_name}.txt
+          done
+          echo "Pod logs saved to recipes/pod-logs/"
+          # Get kubernetes events and save to file
+          kubectl get events -n $namespace > recipes/pod-logs/events.txt
+      - name: Upload Terraform recipe publishing logs 
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: recipes-pod-logs
+          path: recipes/pod-logs
+          retention-days: 30
```

Comment:
What is the default value for retention? Wondering if we should this explicitly for other artifacts as well.

Reply:
![image](https://github.com/radius-project/radius/assets/54363786/376b294c-f7ab-488e-8c09-fb62ba38e33b)
Default is 90

Reply:
Thanks Aaron. Not sure why the artifacts are expiring sooner than 90 days https://github.com/radius-project/radius/actions/runs/7484284145

@sk593 I think 30 days is the right limit for tf recipe pod logs given that we don't use them often. We can review other artifacts separately.

Reply:
> @sk593 I think 30 days is the right limit for tf recipe pod logs given that we don't use them often. We can review other artifacts separately.

Agreed, I don't think 90 days is necessary since this is just for the recipe uploads

Reply:
Maybe runs have different expiration days and the logs get archived somewhere else? Just thinking out loud... 🤣

Reply:
Ah just realized the repo had a retention period (7 days) which was overriding the org-level setting of 90 days. Just updated the repo to 90 days. We can remove this step-level retention period.

Reply:
Ah thanks @AaronCrawfis, good to know, this will be helpful for investigations. Do we have the ability to keep eye on the storage limits, if in case we start hitting them due to this increased retention? In which case we may need to lower it below 90 days.